confidential-data-hub/hub/src/config.rs

```diff
@@ -0,0 +1,404 @@
+// Copyright (c) 2024 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{env, fs, path::Path};
+
+use anyhow::*;
+use attestation_agent::config::aa_kbc_params::AaKbcParams;
+use config::{Config, File};
+use log::{debug, info};
+use serde::Deserialize;
+
+cfg_if::cfg_if! {
+    if #[cfg(feature = "ttrpc")] {
+        const DEFAULT_CDH_SOCKET_ADDR: &str = "unix:///run/confidential-containers/cdh.sock";
+    } else {
+        const DEFAULT_CDH_SOCKET_ADDR: &str = "127.0.0.1:50000";
+    }
+}
+
+#[derive(Clone, Deserialize, Debug, PartialEq)]
+pub struct KbsConfig {
+    pub name: String,
+
+    pub url: String,
+
+    pub kbs_cert: Option<String>,
+}
+
+impl KbsConfig {
+    fn new() -> Result<Self> {
+        debug!("Try to get kbc and url from env and kernel commandline.");
+        let aa_kbc_params =
+            AaKbcParams::new().context("failed to read aa_kbc_params to initialize KbsConfig")?;
+        Ok(KbsConfig {
+            name: aa_kbc_params.kbc,
+            url: aa_kbc_params.uri,
+            kbs_cert: None,
+        })
+    }
+}
+
+#[derive(Clone, Deserialize, Debug, PartialEq)]
+pub struct Credential {
+    pub resource_uri: String,
+    pub path: String,
+}
+
+#[derive(Clone, Deserialize, Debug, PartialEq)]
+pub struct ImageConfiguration {
+    /// Whether to enable image security validation
+    ///
+    /// This defaults to `false`.
+    pub security_validate: bool,
+
+    /// The flag to control whether we use registry credential file to
+    /// access the image registries.
+    ///
+    /// This defaults to `false`.
+    pub use_registry_credentials: bool,
+
+    /// Maximum number of concurrent downloads to perform during image pull.
+    ///
+    /// This defaults to [`image_rs::config::DEFAULT_MAX_CONCURRENT_DOWNLOAD`].
+    pub max_concurrent_download: usize,
+
+    /// sigstore config file for simple signing
+    ///
+    /// This defaults to [`image_rs::config::SIG_STORE_CONFIG_DEFAULT_FILE`].
+    pub sigstore_config_uri: String,
+
+    /// Path to `Policy.json`
+    ///
+    /// This defaults to [`image_rs::config::POLICY_FILE_PATH`].
+    /// The policy follows the format of
+    /// <https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md>.
+    ///
+    /// At the same time, some enhencements based on CoCo is used, that is the
+    /// keyPath field can be filled with a KBS URI like `kbs:///default/key/1`
+    pub policy_uri: String,
+
+    /// Path to the auth file
+    ///
+    /// This defaults to [`image_rs::config::AUTH_FILE_PATH`].
+    pub auth_uri: String,
+
+    /// Proxy that will be used to pull image
+    pub https_proxy: String,
+
+    /// No proxy env that will be used to pull image
+    pub no_proxy: String,
+
+    /// The path to store the pulled data
+    pub work_dir: Option<String>,
```

Comment:
Will this ever be used by someone using CoCo?

Reply:
No. But CDH can work for cases like directly inside IaaS platform rather than CoCo. This would help users to set different workdir which is to store the downloaded layers.

Reply:
@Xynnn007 this is for the case where we can embed image layers in the rootfs for very large images right ?

Reply:
Yes, very large images is typical use case. Also beyond CoCo, CDH could has this configured.

Reply:
@bpradipt If needed, I think you can raise an issue to record this. After these technique preparations are made, we can edit an best practice document for this